mutiny-core/src/federation.rs

```diff
@@ -1087,6 +1321,277 @@ where
+                                labels: labels.clone(),
+                            };
+
+                            match persist_transaction_details(&storage, &updated_transaction_details) {
+                                Ok(_) => {
+                                    log_info!(logger, "Transaction updated");
+                                },
+                                Err(e) => {
+                                    log_error!(logger, "Error updating transaction: {e}");
+                                },
+                            }
+
+                            // we need to get confirmations for this txid and update
+                            subscribe_onchain_confirmation_check(storage.clone(), esplora.clone(), txid, updated_transaction_details, stop, logger.clone()).await;
+
+                            break
+                        },
+                        WithdrawState::Failed(e) => {
+                            log_error!(logger, "Transaction failed: {e}");
+
+                            // Delete the pending tx if it failed
+                            match delete_transaction_details(&storage, internal_id) {
+                                Ok(_) => {
+                                    log_info!(logger, "Transaction deleted");
+                                },
+                                Err(e) => {
+                                    log_error!(logger, "Error deleting transaction: {e}");
+                                },
+                            }
+
+                            break;
+                        },
+                    }
+                }
+            }
+            _ = delay_fut => {
+                if timeout.is_none() {
+                    if stop.load(Ordering::Relaxed)  {
+                        break;
+                    }
+                } else {
+                    log_debug!(
+                        logger,
+                        "Timeout reached, exiting loop for on chain tx",
+                    );
+                    break;
+                }
+            }
+        }
+    }
+    log_trace!(logger, "Done with stream outcome",);
+}
+
+async fn subscribe_onchain_confirmation_check<S: MutinyStorage>(
+    storage: S,
+    esplora: Arc<AsyncClient>,
+    txid: Txid,
+    mut transaction_details: TransactionDetails,
+    stop: Arc<AtomicBool>,
+    logger: Arc<MutinyLogger>,
+) {
+    spawn(async move {
+        loop {
+            if stop.load(Ordering::Relaxed) {
+                break;
+            };
+
+            match esplora.get_tx_status(&txid).await {
+                Ok(s) => {
+                    if s.confirmed {
+                        log_info!(logger, "Transaction confirmed");
+                        transaction_details.confirmation_time = ConfirmationTime::Confirmed {
+                            height: s.block_height.expect("confirmed"),
+                            time: now().as_secs(),
```

Comment:
should use the block's timestamp here if that's available, that's what BDK does and will keep things consistent